Bottom: 2aec7b590ed2c55b869c6663475e388c43101df5
Top:    f99078587eb5f60233498f63cf2e9c7929d4b527
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-14 21:59:49 +0000

Remove HID->set_draw_xor() API

This is only used for drawing the reference mark and objects attached
to the cursor. Rather than have the core setup the XOR drawing mode,
let the GUI set what it wants.

XXX: GTK HID's GDK RENDERER AND LESSTIF HID NOT UPDATED YET


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 937e506..191295d 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1174,7 +1174,6 @@ InitCrosshair (void)
   Crosshair.GC = gui->graphics->make_gc ();
 
   gui->graphics->set_color (Crosshair.GC, Settings.CrosshairColor);
-  gui->graphics->set_draw_xor (Crosshair.GC, 1);
   gui->graphics->set_line_cap (Crosshair.GC, Trace_Cap);
   gui->graphics->set_line_width (Crosshair.GC, 1);
 
diff --git a/src/hid.h b/src/hid.h
index d7974e5..c64c3bd 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -266,7 +266,6 @@ typedef enum
        line style is helpful.  */
     void (*set_line_cap) (hidGC gc, EndCapStyle style);
     void (*set_line_width) (hidGC gc, Coord width);
-    void (*set_draw_xor) (hidGC gc, int xor_);
 
     /* Blends 20% or so color with 80% background.  Only used for
        assembly drawings so far. */
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index f49bc19..824cbaa 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -200,11 +200,6 @@ batch_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-batch_set_draw_xor (hidGC gc, int xor_set)
-{
-}
-
-static void
 batch_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2)
 {
 }
@@ -351,7 +346,6 @@ hid_batch_init ()
   batch_hid.set_color             = batch_set_color;
   batch_hid.set_line_cap          = batch_set_line_cap;
   batch_hid.set_line_width        = batch_set_line_width;
-  batch_hid.set_draw_xor          = batch_set_draw_xor;
   batch_hid.draw_line             = batch_draw_line;
   batch_hid.draw_arc              = batch_draw_arc;
   batch_hid.draw_rect             = batch_draw_rect;
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 1a9cdb2..e19eb7b 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -91,11 +91,6 @@ extents_set_line_width (hidGC gc, Coord width)
   gc->width = width;
 }
 
-static void
-extents_set_draw_xor (hidGC gc, int xor_)
-{
-}
-
 #define PEX(x,w) if (box.X1 > (x)-(w)) box.X1 = (x)-(w); \
 	if (box.X2 < (x)+(w)) box.X2 = (x)+(w)
 #define PEY(y,w) if (box.Y1 > (y)-(w)) box.Y1 = (y)-(w); \
@@ -186,7 +181,6 @@ hid_extents_init (void)
   extents_graphics.set_color      = extents_set_color;
   extents_graphics.set_line_cap   = extents_set_line_cap;
   extents_graphics.set_line_width = extents_set_line_width;
-  extents_graphics.set_draw_xor   = extents_set_draw_xor;
   extents_graphics.draw_line      = extents_draw_line;
   extents_graphics.draw_arc       = extents_draw_arc;
   extents_graphics.draw_rect      = extents_draw_rect;
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 0792df9..47bd421 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -105,12 +105,6 @@ nogui_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-nogui_set_draw_xor (hidGC gc, int xor_)
-{
-  CRASH;
-}
-
-static void
 nogui_set_draw_faded (hidGC gc, int faded)
 {
 }
@@ -454,6 +448,21 @@ common_nogui_init (HID *hid)
   hid->invalidate_all =       nogui_invalidate_all;
   hid->set_layer =            nogui_set_layer;
   hid->end_layer =            nogui_end_layer;
+<<<<<<< current
+=======
+  hid->make_gc =              nogui_make_gc;
+  hid->destroy_gc =           nogui_destroy_gc;
+  hid->use_mask =             nogui_use_mask;
+  hid->set_color =            nogui_set_color;
+  hid->set_line_cap =         nogui_set_line_cap;
+  hid->set_line_width =       nogui_set_line_width;
+  hid->set_draw_faded =       nogui_set_draw_faded;
+  hid->draw_line =            nogui_draw_line;
+  hid->draw_arc =             nogui_draw_arc;
+  hid->draw_rect =            nogui_draw_rect;
+  hid->fill_circle =          nogui_fill_circle;
+  hid->fill_polygon =         nogui_fill_polygon;
+>>>>>>> patched
   hid->fill_pcb_polygon =     nogui_fill_pcb_polygon;
   hid->fill_pcb_pad =         nogui_fill_pcb_pad;
   hid->thindraw_pcb_pad =     nogui_thindraw_pcb_pad;
@@ -496,7 +505,6 @@ common_nogui_graphics_init (HID_DRAW_API *graphics)
   graphics->set_color =       nogui_set_color;
   graphics->set_line_cap =    nogui_set_line_cap;
   graphics->set_line_width =  nogui_set_line_width;
-  graphics->set_draw_xor =    nogui_set_draw_xor;
   graphics->set_draw_faded =  nogui_set_draw_faded;
   graphics->draw_line =       nogui_draw_line;
   graphics->draw_arc =        nogui_draw_arc;
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 4466529..2e316eb 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1233,12 +1233,6 @@ gcode_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-gcode_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 gcode_set_draw_faded (hidGC gc, int faded)
 {
 }
@@ -1601,7 +1595,6 @@ hid_gcode_init ()
   gcode_graphics.set_color      = gcode_set_color;
   gcode_graphics.set_line_cap   = gcode_set_line_cap;
   gcode_graphics.set_line_width = gcode_set_line_width;
-  gcode_graphics.set_draw_xor   = gcode_set_draw_xor;
   gcode_graphics.set_draw_faded = gcode_set_draw_faded;
   gcode_graphics.draw_line      = gcode_draw_line;
   gcode_graphics.draw_arc       = gcode_draw_arc;
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index b884c51..54ba61e 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -52,7 +52,6 @@ static void gerber_use_mask (enum mask_mode mode);
 static void gerber_set_color (hidGC gc, const char *name);
 static void gerber_set_line_cap (hidGC gc, EndCapStyle style);
 static void gerber_set_line_width (hidGC gc, Coord width);
-static void gerber_set_draw_xor (hidGC gc, int _xor);
 static void gerber_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void gerber_draw_arc (hidGC gc, Coord cx, Coord cy, Coord width, Coord height, Angle start_angle, Angle delta_angle);
 static void gerber_draw_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
@@ -933,12 +932,6 @@ gerber_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-gerber_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc, int radius)
 {
   if (radius)
@@ -1291,7 +1284,6 @@ hid_gerber_init ()
   gerber_graphics.set_color      = gerber_set_color;
   gerber_graphics.set_line_cap   = gerber_set_line_cap;
   gerber_graphics.set_line_width = gerber_set_line_width;
-  gerber_graphics.set_draw_xor   = gerber_set_draw_xor;
   gerber_graphics.draw_line      = gerber_draw_line;
   gerber_graphics.draw_arc       = gerber_draw_arc;
   gerber_graphics.draw_rect      = gerber_draw_rect;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 1913210..c05ed35 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -484,16 +484,6 @@ ghid_set_line_width (hidGC gc, Coord width)
   gc->width = width;
 }
 
-
-void
-ghid_set_draw_xor (hidGC gc, int xor)
-{
-  /* NOT IMPLEMENTED */
-
-  /* Only presently called when setting up a crosshair GC.
-   * We manage our own drawing model for that anyway. */
-}
-
 void
 ghid_set_draw_faded (hidGC gc, int faded)
 {
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 4e8f0fe..2391a21 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2189,7 +2189,6 @@ hid_gtk_init ()
   ghid_graphics.set_color           = ghid_set_color;
   ghid_graphics.set_line_cap        = ghid_set_line_cap;
   ghid_graphics.set_line_width      = ghid_set_line_width;
-  ghid_graphics.set_draw_xor        = ghid_set_draw_xor;
   ghid_graphics.draw_line           = ghid_draw_line;
   ghid_graphics.draw_arc            = ghid_draw_arc;
   ghid_graphics.draw_rect           = ghid_draw_rect;
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index af719fd..41a12b7 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -476,7 +476,6 @@ void ghid_use_mask (enum mask_mode mode);
 void ghid_set_color (hidGC gc, const char *name);
 void ghid_set_line_cap (hidGC gc, EndCapStyle style);
 void ghid_set_line_width (hidGC gc, Coord width);
-void ghid_set_draw_xor (hidGC gc, int _xor);
 void ghid_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 void ghid_draw_arc (hidGC gc, Coord cx, Coord cy, Coord xradius, Coord yradius,
                     Angle start_angle, Angle delta_angle);
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 04d9f6c..17f2b4b 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4063,7 +4063,6 @@ hid_lesstif_init ()
   lesstif_hid.set_color               = lesstif_set_color;
   lesstif_hid.set_line_cap            = lesstif_set_line_cap;
   lesstif_hid.set_line_width          = lesstif_set_line_width;
-  lesstif_hid.set_draw_xor            = lesstif_set_draw_xor;
   lesstif_hid.draw_line               = lesstif_draw_line;
   lesstif_hid.draw_arc                = lesstif_draw_arc;
   lesstif_hid.draw_rect               = lesstif_draw_rect;
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index dfaf3f5..9bac444 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -820,12 +820,6 @@ nelma_set_line_width(hidGC gc, Coord width)
 }
 
 static void
-nelma_set_draw_xor(hidGC gc, int xor_)
-{
-	;
-}
-
-static void
 nelma_set_draw_faded(hidGC gc, int faded)
 {
 }
@@ -1063,7 +1057,6 @@ hid_nelma_init()
   nelma_graphics.set_color      = nelma_set_color;
   nelma_graphics.set_line_cap   = nelma_set_line_cap;
   nelma_graphics.set_line_width = nelma_set_line_width;
-  nelma_graphics.set_draw_xor   = nelma_set_draw_xor;
   nelma_graphics.set_draw_faded = nelma_set_draw_faded;
   nelma_graphics.draw_line      = nelma_draw_line;
   nelma_graphics.draw_arc       = nelma_draw_arc;
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 848bac7..c1c08b7 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1464,12 +1464,6 @@ png_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-png_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc)
 {
   int need_brush = 0;
@@ -1829,7 +1823,6 @@ hid_png_init ()
   png_graphics.set_color      = png_set_color;
   png_graphics.set_line_cap   = png_set_line_cap;
   png_graphics.set_line_width = png_set_line_width;
-  png_graphics.set_draw_xor   = png_set_draw_xor;
   png_graphics.draw_line      = png_draw_line;
   png_graphics.draw_arc       = png_draw_arc;
   png_graphics.draw_rect      = png_draw_rect;
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index f09c601..3542ab3 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -39,7 +39,6 @@ static void eps_use_mask (enum mask_mode mode);
 static void eps_set_color (hidGC gc, const char *name);
 static void eps_set_line_cap (hidGC gc, EndCapStyle style);
 static void eps_set_line_width (hidGC gc, Coord width);
-static void eps_set_draw_xor (hidGC gc, int _xor);
 static void eps_draw_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void eps_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void eps_draw_arc (hidGC gc, Coord cx, Coord cy, Coord width, Coord height, Angle start_angle, Angle delta_angle);
@@ -519,12 +518,6 @@ eps_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-eps_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc)
 {
   if (linewidth != gc->width)
@@ -693,7 +686,6 @@ hid_eps_init ()
   eps_graphics.set_color      = eps_set_color;
   eps_graphics.set_line_cap   = eps_set_line_cap;
   eps_graphics.set_line_width = eps_set_line_width;
-  eps_graphics.set_draw_xor   = eps_set_draw_xor;
   eps_graphics.draw_line      = eps_draw_line;
   eps_graphics.draw_arc       = eps_draw_arc;
   eps_graphics.draw_rect      = eps_draw_rect;
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index d665e6c..11f8bb6 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1067,12 +1067,6 @@ ps_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-ps_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 ps_set_draw_faded (hidGC gc, int faded)
 {
   gc->faded = faded;
@@ -1509,7 +1503,6 @@ void ps_ps_graphics_init (HID_DRAW_API *graphics)
   graphics->set_color          = ps_set_color;
   graphics->set_line_cap       = ps_set_line_cap;
   graphics->set_line_width     = ps_set_line_width;
-  graphics->set_draw_xor       = ps_set_draw_xor;
   graphics->set_draw_faded     = ps_set_draw_faded;
   graphics->draw_line          = ps_draw_line;
   graphics->draw_arc           = ps_draw_arc;
